### ISSUE TITLE
Update TooWaBlue, Deutsch locale, Images

### ISSUE THREAD

**Issue by TooWaBoo** [Beep6581/RawTherapee#3247]
Deutsch locale update
[Deutsch.txt](https://github.com/Beep6581/RawTherapee/files/210516/Deutsch.txt)

TooWaBlue v1.19.3 update
Changed slider knop a bit.
Add a separate color defenition to the change me section for the the slider knop.
[TooWaBlue.css.txt](https://github.com/Beep6581/RawTherapee/files/224498/TooWaBlue.css.txt)

Installed the new RT 4.2.915 and the disabled checkbox and radio button are to dark.
I've modified the included SVG by hand to fit font color/brightness and I changed the stroke-width for the square of the checkbox to make it look a bit smoother. THE PNG were created by an online convertertool from SVG to PNG.

The opacity is set to 70%, but what I did is to double the vector image in the SVG.
opacity0.7 + opacity0.7 = opacity1.4 -> opacity = 1 -  Not realy, but close to 1  :-)

RT is using colored images, so to set a specific color can't be a problem. I changed the color inside the SVG to the grey to fit the font color/brightness.

You can download the PNG and SVG here: http://filebin.net/xxqhufldl0

Left New, Right old
![698c700e-fc25-11e5-8e18-f634d95109a9](https://cloud.githubusercontent.com/assets/13800920/14390589/23b4149e-fdb8-11e5-9ab8-8dddcde124ca.png)

**Comment by TooWaBoo** [comment]
ping

**Comment by Beep6581** [comment]
The CSS file uses many colors, and states them using two notations: hex and RGB(A) 0-255. Is that necessary? It's fine if it is, but it would be helpful to simplify it if possible by cutting down on the color variations (alpha=0.1 vs 0.12, is that really a visible difference?) and using one style. I had a hard time finding which color you actually use for the "normal" vs "-disabled" text.

> The opacity is set to 70%, but what I did is to double the vector image in the SVG.
opacity0.7 + opacity0.7 = opacity1.4 -> opacity = 1 - Not realy, but close to 1 :-)

That's nice :) But it's a hack and we should not do that. The problem was that we use a script to generate PNG icons from SVG files for all icons in RawTherapee, and the whole thing relies on using a specific color and specific opacity in the SVG so that these can be replaced using a stream editor to get a consistent style. But as TooWaBlue is your theme and it needs your images to match your colors, I abandoned that script and wrote a new, simple one to handle theme-specific images.

I edited the SVG icons to use colors which match your text colors (active=`#b4b4b4`, disabled=`#808080`, is that correct?), and added `tools/generateThemeSVG2PNG` which generates PNG icons from the SVG files in the right folders. To use it, just
`cd rtdata/themes/images/`
and run
`../../../tools/generateThemeSVG2PNG`

> I changed the stroke-width for the square of the checkbox to make it look a bit smoother

I didn't keep this change because it wasn't consistent. The square had a stroke width of 1.2 pixels while the radio button and in your screenshots the frames use a width of 1 pixel.

Please take a look at the files I committed. If you feel that the color needs changing, we can do that easily now. If you still want to change the stroke width, we can do that too, but let's try to do that consistently, so if we do it for `checkbox-*` we should also do it for `radio-*`.

**Comment by TooWaBoo** [comment]
I've changed all color notations to rgb(a).
[TooWaBlue1.19.4.css.txt](https://github.com/Beep6581/RawTherapee/files/232908/TooWaBlue1.19.4.css.txt)

> (alpha=0.1 vs 0.12, is that really a visible difference?)

Yes, it is, speciality on gradients.

> But it's a hack and we should not do that.

Maybe it's not common, but it's regulare function and you can do that in "inkscape" too.

>  (active=#b4b4b4, disabled=#808080, is that correct?)

That's correct.

> The square had a stroke width of 1.2 pixels while the radio button and in your screenshots the frames use a width of 1 pixel.

The radio has a stroke width of 1, but it looks thicker than 1 pixel, 'cause of the antialiasing. That's why I increased the stroke width of the checkbox square to match the radio. If there is'nt a technical reason to use a stroke width of 1.2 only for the checkbox square, I want to keep it.

> added tools/generateThemeSVG2PNG which generates PNG icons from the SVG files in the right folders. To use it, just
> `cd rtdata/themes/images/`
> and run
> `../../../tools/generateThemeSVG2PNG`

I've no clue, how I can do that in Windows.
Is it correct, that the PNGs in  rtdata/themes/images/ are no PNGs?

**Comment by Beep6581** [comment]
I updated the checkbox SVGs and re-generated the PNGs to use a stroke width of 1.2.

> Is it correct, that the PNGs in rtdata/themes/images/ are no PNGs?

The PNGs in `rtdata/themes/images/twb` are ordinary PNGs, they are automatically generated from the SVGs in `rtdata/themes/images/svg/twb`, that's why it's important that this is all scriptable, as it now is.

I committed the new theme and updated SVG+PNG icons. OK to close?

P.S. "knop" should be "knob"
